db/python/layers/sequence.py

```diff
@@ -86,7 +86,9 @@ async def get_sequences_for_sample_ids(
         """
         projects, sequences = await self.seqt.get_sequences_by(sample_ids=sample_ids)
 
-        if check_project_ids:
+        if (
+            check_project_ids and projects and sequences
```

Comment:
This should fail if there are sequences but no projects
```suggestion
            check_project_ids and sequences
```

Reply:
Fixed